### PR TITLE
Update to v0.15.0

### PR DESCRIPTION
**Pull request by sigma-andex**

### CODE REVIEW

**Comment by sigma-andex** [comment]
@natefaubion this is now updated to the v0.15 relase

**Comment by natefaubion** [comment]
I appreciate this PR, however, isn't it the case that `variant` already works fine on 0.15?

**Comment by thomashoneyman** [comment]
As far as bower users are concerned, they’ll need a major version bump with all those dependencies moved up a major version. But variant is fine in the package sets with the code as-is iirc.

**Comment by natefaubion** [comment]
This PR does not update bower dependencies.

**Comment by sigma-andex** [comment]
Are you sure about this? If I use the master version I get the following error:
```
➜ spago build
[info] Installing 1 dependencies.
[info] Searching for packages cache metadata..
[info] Recent packages cache metadata found, using it..
[info] Installing "variant"
[info] Installation complete.
[ 1 of 17] Compiling Data.Variant.Internal
[ 2 of 17] Compiling Data.Variant
[ 3 of 17] Compiling Data.Functor.Variant
[1/2 TypesDoNotUnify] .spago/variant/master/src/Data/Functor/Variant.purs:179:45

  179      VariantFRep v | v.type == reflectSymbol p → f v.value
                                                   ^
  
  Could not match type
  
    proxy1
  
  with type
  
    Proxy
  
  while trying to match type proxy1 sym2
    with type Proxy @Symbol t0
  while checking that expression p
    has type Proxy @Symbol t0
  while checking the type of a guard clause
  in value declaration on
  
  where proxy1 is a rigid type variable
          bound at (line 0, column 0 - line 0, column 0)
        sym2 is a rigid type variable
          bound at (line 0, column 0 - line 0, column 0)
        t0 is an unknown type

[2/2 TypesDoNotUnify] .spago/variant/master/src/Data/Variant.purs:94:44

  94      VariantRep v | v.type == reflectSymbol p → f v.value
                                                 ^
  
  Could not match type
  
    proxy1
  
  with type
  
    Proxy
  
  while trying to match type proxy1 sym2
    with type Proxy @Symbol t0
  while checking that expression p
    has type Proxy @Symbol t0
  while checking the type of a guard clause
  in value declaration on
  
  where proxy1 is a rigid type variable
          bound at (line 0, column 0 - line 0, column 0)
        sym2 is a rigid type variable
          bound at (line 0, column 0 - line 0, column 0)
        t0 is an unknown type

           Src   Lib   All
Warnings   0     0     0  
Errors     0     2     2  
[error] Failed to build.
```

**Comment by sigma-andex** [comment]
I have the same problem in routing-duplex btw

**Comment by JordanMartinez** [comment]
@sigma-andex That's because you need to drop the `forall proxy. proxy sym` with just `Proxy sym`. For example, this (https://github.com/natefaubion/purescript-variant/blob/master/src/Data/Variant.purs#L56) becomes:
```purs
import Type.Proxy (Proxy(..))

inj
  ∷ ∀ sym a r1 r2
  . R.Cons sym a r1 r2
  ⇒ IsSymbol sym
  ⇒ Proxy sym
  → a
  → Variant r2
inj p value = coerceV $ VariantRep { type: reflectSymbol p, value }
  where
  coerceV ∷ VariantRep a → Variant r2
  coerceV = unsafeCoerce
```
Have you looked through the [migration guide](https://github.com/purescript/documentation/blob/master/migration-guides/0.15-Migration-Guide.md#changes-affecting-multiple-libraries=)? :wink: If that section needs to be clearer, we can update it.

**Comment by JordanMartinez** [comment]
I just checked this PR out locally and it builds and tests fine for me. @sigma-andex Have you done ran 
```
rm -rf bower_components/ output/
bower cache clean
bower install
pulp test
```

**Comment by sigma-andex** [comment]
The PR builds with my change. I mean this error occurs if variant stays unchanged.

**Comment by JordanMartinez** [comment]
Ah!

And on that note, one workaround is to specify the `resolutions` field in the `bower.json` file. I haven't published a library with that, and I'm not sure whether we should encourage that.

File: spago.dhall [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- Dhall language tour: https://docs.dhall-lang.org/tutorials/Language-Tour.html
+
+When creating a new Spago project, you can use
+`spago init --no-comments` or `spago init -C`
+to generate this file without the comments in this block.
+-}
+{ name = "variant"
+, dependencies =
+  [ "assert"
+  , "console"
```
Comment (natefaubion) [inline spago.dhall:16]: I'm getting warnings about `console` from `spago`. Seems to be unused?

**Comment by natefaubion** [comment]
Thanks! I fixed it up in main.

**Pull request merged**